### PR TITLE
Use BFD for travis tests

### PR DESCRIPTION
**Pull request by isuruf**
`binutils-dev` was added to apt-whitelist of Travis-CI , so now we can use it in Travis-CI
https://github.com/travis-ci/travis-ci/issues/4113

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,28 +24,30 @@ env:
   ## Out of tree builds (default):
   # Debug build
   - BUILD_TYPE="Debug"
-  # Debug build (with SYMENGINE_ASSERT and Teuchos::RCP)
-  - BUILD_TYPE="Debug" WITH_SYMENGINE_ASSERT="yes" WITH_SYMENGINE_RCP="no"
```
Comment (certik) [inline .travis.yml]: Why did you remove `WITH_SYMENGINE_RCP="no"` in all tests? We want to keep it, as this makes sure that all tests pass with the strict checking, thus ensuring the code is safe.

Comment (isuruf) [inline .travis.yml]: I thought it was default with `BUILD_TYPE="Debug"`

Comment (certik) [inline .travis.yml]: Ah, that's right, I forgot about that. Yes, it is the default. So all is good.

File: .travis.yml [hunk 2]
```diff
@@ -23,28 +24,30 @@ env:
...
+  # Debug build (with BFD and SYMENGINE_ASSERT) with PRIMESIEVE
+  - BUILD_TYPE="Debug" WITH_BFD="yes" WITH_SYMENGINE_ASSERT="yes" WITH_PRIMESIEVE="yes"
+  # Debug build (with BFD and SYMENGINE_ASSERT) with Arb
+  - BUILD_TYPE="Debug" WITH_BFD="yes" WITH_SYMENGINE_ASSERT="yes" WITH_ARB="yes"
+  # Debug build (with BFD and SYMENGINE_ASSERT) with MPC
+  - BUILD_TYPE="Debug" WITH_BFD="yes" WITH_SYMENGINE_ASSERT="yes" WITH_MPC="yes"
+  # Debug build shared lib (with BFD and SYMENGINE_ASSERT)
+  - BUILD_TYPE="Debug" WITH_BFD="yes" WITH_SYMENGINE_ASSERT="yes" BUILD_SHARED_LIBS="yes"
+  # Release build (with BFD)
+  - WITH_BFD="yes"
```
Comment (isuruf) [inline .travis.yml]: This was a test previously in there, but I had accidentally deleted it when removing `WITH_BFD="yes"`

Comment (certik) [inline .travis.yml]: That's fine.

I think we should enable `WITH_SYMENGINE_ASSERT="yes"` automatically in Debug mode.

File: .travis.yml [hunk 3]
```diff
@@ -88,7 +92,7 @@ matrix:
           - clang
           - libstdc++-4.8-dev
           - libgmp-dev
-    - env: BUILD_TYPE="Debug" WITH_SYMENGINE_ASSERT="yes" WITH_SYMENGINE_RCP="no" WITH_PYTHON="yes" PYTHON_VERSION="2.7"
+    - env: BUILD_TYPE="Debug" WITH_BFD="yes" WITH_SYMENGINE_ASSERT="yes" WITH_PYTHON="yes" PYTHON_VERSION="2.7"
```
Comment (certik) [inline .travis.yml]: I think you'll have to remove WITH_BFD on a Mac, as Travis fails due to this. Looks like the bfd library is there but not `link.h`. There might be some other way to implement it on a Mac, and we can look into it later, as there are many people who use OS X as their main development platform. For now I would just disable it.

Comment (isuruf) [inline .travis.yml]: Removed it in f572a9f

File: .travis.yml (hunk 2 above)
Comment (isuruf) [inline .travis.yml]: Yes, we should do that

**Comment by certik** [comment]
+1 to merge if tests pass.
